v2/inc/c_logging/log_context.h

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef LOG_CONTEXT_H
+#define LOG_CONTEXT_H
+
+#ifdef __cplusplus
+#include <cstdint>
+#include <cstdio>
+#include <cstdlib>
+#else
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#endif
+
+#include "macro_utils/macro_utils.h"
+
+// This code is intentionally a spike only to showcase an implementation
+// Code with tests shall be written after comments on the general idea are received
+
+typedef struct LOG_CONTEXT_TAG
+{
+    char* context_string;
```

Comment:
>context_string [](http://example.com/codeflow?start=10&length=14)

aha, so the context is an unstructured string.

Reply:
Now the fields in it have the concept of a type attached to them